src/krb5_plugin/passkey/passkey_clpreauth.c

```diff
@@ -52,14 +52,45 @@ sss_passkeycl_prep_questions(krb5_context context,
 {
     struct sss_passkey_message *message;
     char *question = NULL;
+    char **realms = NULL;
     krb5_error_code ret;
+    size_t r = 0;
 
     message = sss_passkey_message_decode_padata(pa_data);
     if (message == NULL) {
         ret = ENOMEM;
         goto done;
     }
 
+    if (message->data.challenge->domain == NULL) {
+        ret = KRB5KDC_ERR_PREAUTH_FAILED;
+        goto done;
+    }
+
+    /* Find a realm that matches the domain from the challenge */
+    ret = krb5_get_host_realm(context,
+                              message->data.challenge->domain,
+                              &realms);
+    if (ret || ((realms == NULL) || (realms[0] == NULL))) {
+        ret = KRB5KDC_ERR_PREAUTH_FAILED;
```

Comment:
Thanks. Is it about NULL or "" as an empty string?

Reply:
Hi,

I see
```
(gdb) p realms
$2 = (char **) 0x409f30
(gdb) p realms[0]
$3 = 0x405590 ""
(gdb) p realms[1]
$4 = 0x0
```
and I think this is in agreement the description in the header file
```
 * Fill in @a realmsp with a pointer to a null-terminated list of realm names.
 * If there are no known realms for the host, a list containing the referral
 * (empty) realm is returned.
```

bye,
Sumit

Reply:
wonderful. I added a bit more explicit check for the referral realm:
```
    /* Do explicit check for the challenge domain in case
     * we've got back a referral (empty) realm */
    if (strlen(realms[0]) == strlen(KRB5_REFERRAL_REALM)) {
...
```